geoutils/georaster.py

```diff
@@ -95,8 +96,28 @@ def __init__(self, filename, attrs=None, load_data=True, bands=None,
         # Save _masked attribute to be used by self.load()
         self._masked = masked
 
+        # Check number of bands to be loaded
+        # If bands is int, force to be list so that self.data.ndim is 3
+        if bands is None:
+            nbands = self.count
+        elif isinstance(bands, int):
+            nbands = 1
+            bands = (bands,)
+        elif isinstance(bands, collections.abc.Iterable):
+            nbands = len(bands)
+
+        # Downsampled image size
+        if not isinstance(downsampl, int):
+            raise ValueError("downsampl must be of type int")
+        if downsampl == 1:
+            out_shape = (nbands, self.height, self.width)
+        else:
+            down_width = int(np.ceil(self.width/downsampl))
+            down_height = int(np.ceil(self.height/downsampl))
+            out_shape = (nbands, down_height, down_width)
+
         if load_data:
-            self.load()
+            self.load(bands=bands, out_shape=out_shape)
```

Comment:
I see, with out_shape passed down directly to rio.ds.read(); should we leave the default nearest interpolation, or change to bilinear? in some case (e.g. for DEMs), I guess it's still better to downsample using the exact decimal coordinates. Not sure if truly necessary... and would need to add a `downsamp_method` argument to __init__()

Reply:
For me, the main use case is simply for quickly visualizing large rasters. So nearest is probably the fastest. I would not use it for running calculations on it. In that case, we have self.interp etc.
There is still the option to use a different resampling algo using self.load directly.